src/diffusers/loaders/single_file.py

```diff
@@ -12,143 +12,126 @@
+    else:
+        transformers_version = "N/A"
+
+    is_transformers_model = (
+        is_transformers_available()
+        and issubclass(class_obj, PreTrainedModel)
+        and transformers_version >= version.parse("4.20.0")
+    )
+
+    diffusers_module = importlib.import_module(__name__.split(".")[0])
+    is_diffusers_single_file_model = issubclass(class_obj, diffusers_module.FromOriginalModelMixin)
+
+    if is_diffusers_single_file_model:
+        if original_config is None:
+            config = class_obj.load_config(
+                pretrained_model_name_or_path,
+                subfolder=name,
+                local_files_only=local_files_only,
+                **kwargs,
+            )
+        else:
+            config = None
+
+        load_method = getattr(class_obj, "from_single_file")
+        loaded_sub_model = load_method(
+            checkpoint=checkpoint,
+            original_config=original_config,
+            config=config,
             torch_dtype=torch_dtype,
-            model_type=model_type,
-            upcast_attention=upcast_attention,
-        )
-        return unet_components
-
-    if component_name == "vae":
-        scaling_factor = kwargs.get("scaling_factor", None)
-        vae_components = create_diffusers_vae_model_from_ldm(
-            pipeline_class_name,
-            original_config,
-            checkpoint,
-            image_size,
-            scaling_factor,
-            torch_dtype,
-            model_type=model_type,
-        )
-        return vae_components
-
-    if component_name == "scheduler":
-        scheduler_type = kwargs.get("scheduler_type", "ddim")
-        prediction_type = kwargs.get("prediction_type", None)
-
-        scheduler_components = create_scheduler_from_ldm(
-            pipeline_class_name,
-            original_config,
-            checkpoint,
-            scheduler_type=scheduler_type,
-            prediction_type=prediction_type,
-            model_type=model_type,
+            **kwargs,
         )
 
-        return scheduler_components
+    elif is_transformers_model and is_clip_model_in_single_file(class_obj, checkpoint):
+        load_method = create_diffusers_clip_model_from_ldm
+        config = class_obj.config_class.from_pretrained(pretrained_model_name_or_path, subfolder=name, **kwargs)
 
-    if component_name in ["text_encoder", "text_encoder_2", "tokenizer", "tokenizer_2"]:
-        text_encoder_components = create_text_encoders_and_tokenizers_from_ldm(
-            original_config,
-            checkpoint,
-            model_type=model_type,
+        loaded_sub_model = load_method(
+            class_obj,
+            checkpoint=checkpoint,
+            subfolder=name,
+            config=config,
+            torch_dtype=torch_dtype,
+            local_files_only=local_files_only,
+            **kwargs,
+        )
+
+    else:
+        load_method = getattr(class_obj, "from_pretrained")
+        loaded_sub_model = load_method(
```

Comment:
Hmm so single file checkpoints don't contain schedulers. This warning would then always show up when loading the pipe components?

Reply:
maybe just modules then? I will let you decide since you are the expert here, 

the use case I have in mind that we need to warn us that the checkpoint contains weights for a model class that we do not support `from_single_file` yet; in that case, I think we would just load the pre-trained model from the hub - is this possible to happen? would user get any warnings?